testing/python/metal/test_metal_adapter.py

```
@@ -0,0 +1,1866 @@
+# ---------------------------------------------------------------------------
+# adapter expression §4: multi-runtime-scalar packing (≥2 runtime scalar kernel
+# arguments).  The Metal codegen packs every scalar parameter into a single
+# ``args_t`` struct at ``buffer(num_buffer)``, but the
+# ``torch.mps.compile_shader`` launcher binds each positional argument to
+# its own buffer slot, so pre-fix only the first scalar landed in the struct
+# and the rest silently read stale bytes (``kern(a, 5, 7)`` -> ``a + 5000``
+# instead of ``a + 5007``).  The adapter must pack all runtime scalars into
+# one buffer reproducing the struct layout.  All cases run on real MPS.
+# ---------------------------------------------------------------------------
+@T.prim_func
+def adapter_expr_two_scalar_middle(A: T.Tensor((64,), "float32"), I: T.int32, J: T.int32, OUT: T.Tensor((64,), "float32")):
```

Ruff (0.16.1)

[error] Ambiguous variable name: `I`

(E741)